api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -159,6 +165,44 @@ private void validateStopDatetime(Visit visit, Visit otherVisit, Errors errors)
 			errors.rejectValue("stopDatetime", "Visit.visitCannotContainAnotherVisitOfTheSamePatient", messageBuilder
 			        .toString());
 		}
-		
+
+	}
+
+	private void validateStartDateTimeToBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
+			if (visit.getStartDatetime() != null && patient.getBirthdate() != null) {
+				if (!patient.getBirthdateEstimated() && visit.getStartDatetime().before(patient.getBirthdate())) {
+					errors.rejectValue("startDatetime", "Visit.startDateCannotBeBeforeBirthdate",
```

Comment:
You can avoid code duplication by creating a local variable for the birthdate, assign the estimated date or the actual date of course depending on if it's estimated vs not and then do the comparison and rejection of the value once